### PR TITLE
Core/Spells: Fix mage Comet Storm

### LINKED ISSUE #23689: spell 153595 comet storm
**Spell::prepare: spell id 153595 source 0 caster 0 customCastFlags 0 mask 2
Sending SMSG_SPELL_START id=153595
CriteriaHandler::UpdateCriteria(CAST_SPELL, 29, 153595, 0, 0) GUID Full: 0x08000400000000000000000000000003 Type: Player Entry: 0 Low: 3 Testa
CriteriaHandler::UpdateCriteria(CAST_SPELL, 29, 153595, 0, 0) GUID Full: 0x08000400000000000000000000000003 Type: Player Entry: 0 Low: 3 Testa
Spell: 153595 Effect: 3
Spell: 153595 Effect: 3
Sending SMSG_SPELL_GO id=153595
Spell: 153595 Effect: 3
CriteriaHandler::UpdateCriteria(CAST_SPELL2, 110, 153595, 0, 0) GUID Full: 0x08000400000000000000000000000003 Type: Player Entry: 0 Low: 3 Testa
CriteriaHandler::UpdateCriteria(CAST_SPELL2, 110, 153595, 0, 0) GUID Full: 0x08000400000000000000000000000003 Type: Player Entry: 0 Low: 3 Testa
Spell: 153595 Effect: 3
Spell ScriptStart spellid 153595 in EffectDummy(0)**

CHANGEME Description of the problem or issue here.

**visual effect only on the character. No damage**

CHANGEME Tell us what happens.
If this is a crash, post the crashlog (upload to https://gist.github.com/).

**fix**

CHANGEME Tell us what should happen instead.

**Mage blood elf. 120 level. Application on mobs. Mana not consumed**

1. CHANGEME Step 1 include entries of affected creatures / items / quests with a link to the relevant wowhead page.  
2. Step 2
3. Step 3

**master**

CHANGEME 3.3.5, master or both

**of 27/07/2019** 

CHANGEME Copy the result of server debug command (if you need to run it from client get prat addon)

**Win 10 x64 1803** CHANGEME OS

### PR DESCRIPTION
**Pull request by aquadeus**
**Changes proposed:**

-  Fix frost talent Comet Storm

**Issues addressed:**

Closes #23689

**Tests performed:**

Tested in-game

### CODE REVIEW

File: src/server/scripts/Spells/spell_mage.cpp [hunk 1]
```diff
@@ -430,6 +432,71 @@ class spell_mage_cold_snap : public SpellScript
     }
 };
 
+// 153595 - Comet Storm (launch)
+class spell_mage_comet_storm : public SpellScript
+{
+    PrepareSpellScript(spell_mage_comet_storm);
+
+    class CometStormEvent : public BasicEvent
```
Comment (Nyr97) [inline src/server/scripts/Spells/spell_mage.cpp:440]: Should the basic event not be outside the class?

Comment (aquadeus) [reply src/server/scripts/Spells/spell_mage.cpp:440]: doesn't matter actually

File: src/server/scripts/Spells/spell_mage.cpp [hunk 2]
```diff
@@ -430,6 +432,71 @@ class spell_mage_cold_snap : public SpellScript
...
+    PrepareSpellScript(spell_mage_comet_storm_damage);
+
+    bool Validate(SpellInfo const* /*spellInfo*/) override
+    {
+        return ValidateSpellInfo ({ SPELL_MAGE_COMET_STORM_DAMAGE });
+    }
+
+    void HandleEffectHitTarget(SpellEffIndex /*effIndex*/)
+    {
+        GetCaster()->CastSpell(Position { GetHitDest()->GetPositionX(), GetHitDest()->GetPositionY(), GetHitDest()->GetPositionZ() }, SPELL_MAGE_COMET_STORM_DAMAGE, true);
```
Comment (mdX7) [inline src/server/scripts/Spells/spell_mage.cpp:491]: just use `*GetHitDest()`

File: src/server/scripts/Spells/spell_mage.cpp [hunk 3]
```diff
@@ -430,6 +432,71 @@ class spell_mage_cold_snap : public SpellScript
...
 };
 
+class CometStormEvent : public BasicEvent
+{
+    public:
+        CometStormEvent(Unit* caster, Position const& dest) : _caster(caster), _dest(dest), _count(0) { }
+
+        bool Execute(uint64 time, uint32 /*diff*/) override
+        {
+            _caster->CastSpell(Position { _dest.GetPositionX() + frand(-3.0f, 3.0f), _dest.GetPositionY() + frand(-3.0f, 3.0f), _dest.GetPositionZ() }, SPELL_MAGE_COMET_STORM_VISUAL, true);
```
Comment (mdX7) [inline src/server/scripts/Spells/spell_mage.cpp:442]: move position to separate variable, makes it easier to read

**Pull request merged**